### PR TITLE
Planning: record optimal speed information in DP search

### PR DESCRIPTION
**Pull request by JasonXu12**

### CODE REVIEW

File: modules/planning/tasks/optimizers/path_time_heuristic/gridded_path_time_graph.cc [hunk 1]
```diff
@@ -143,6 +143,12 @@ Status GriddedPathTimeGraph::Search(SpeedData* const speed_data) {
 Status GriddedPathTimeGraph::InitCostTable() {
   // Time dimension is homogeneous while Spatial dimension has two resolutions,
   // dense and sparse with dense resolution coming first in the spatial horizon
+  constexpr double kDoubleEpsilon = 1.0e-6;  // The least meaningful unit_t_
```
Comment (xiaoxq) [inline modules/planning/tasks/optimizers/path_time_heuristic/gridded_path_time_graph.cc]: static

File: modules/planning/tasks/optimizers/path_time_heuristic/gridded_path_time_graph.cc [hunk 2]
```diff
@@ -143,6 +143,12 @@ Status GriddedPathTimeGraph::Search(SpeedData* const speed_data) {
...
+  if (unit_t_ < kDoubleEpsilon) {
+    // Sanity check for numerical stability
+    const std::string msg = "unit_t is smaller than the kDoubleEpsilon.";
+    AERROR << msg;
```
Comment (xiaoxq) [inline modules/planning/tasks/optimizers/path_time_heuristic/gridded_path_time_graph.cc]: If not reused, put the string after AERROR directly.

File: modules/planning/tasks/optimizers/path_time_heuristic/st_graph_point.cc [hunk 3]
```diff
@@ -71,5 +71,11 @@ void StGraphPoint::SetPrePoint(const StGraphPoint& pre_point) {
   pre_point_ = &pre_point;
 }
 
+const double StGraphPoint::GetOptimalSpeed() const { return optimal_speed_; }
```
Comment (xiaoxq) [inline modules/planning/tasks/optimizers/path_time_heuristic/st_graph_point.cc]: No need to return a const copy, just a copy.

File: modules/planning/tasks/optimizers/path_time_heuristic/gridded_path_time_graph.cc (hunk 1 above)
Comment (JasonXu12) [reply modules/planning/tasks/optimizers/path_time_heuristic/gridded_path_time_graph.cc]: Fixed. Thank you!

File: modules/planning/tasks/optimizers/path_time_heuristic/gridded_path_time_graph.cc (hunk 2 above)
Comment (JasonXu12) [reply modules/planning/tasks/optimizers/path_time_heuristic/gridded_path_time_graph.cc]: Fixed. Thank you!

File: modules/planning/tasks/optimizers/path_time_heuristic/st_graph_point.cc (hunk 3 above)
Comment (JasonXu12) [reply modules/planning/tasks/optimizers/path_time_heuristic/st_graph_point.cc]: Fixed. Thank you!

**Pull request merged**